src/pages/index.js

```diff
@@ -84,26 +104,153 @@ editCardPopup.setEventListeners();
 const userInfo = new UserInfo({
   titleSelector: ".profile__title",
   descriptionSelector: ".profile__description",
+  avatar: ".profile__image",
 });
 
+/*Delete card confirmation*/
+
+const deleteConfirmPopup = new PopupWithConfirmation({
+  popupSelector: "#modal-delete-card",
+  handleFormSubmit: () => {},
+});
+deleteConfirmPopup.setEventListeners();
+
+/*Edit Avatar*/
+
+const avatarProfilePopup = new PopupWithForm(
+  "#profile-avatar-modal",
+  avatarSubmit
+);
+avatarProfilePopup.setEventListeners();
+
+const EditAvatarBtn = document.querySelector(".profile__avatar-button");
+EditAvatarBtn.addEventListener("click", () => {
+  avatarProfilePopup.open();
+});
 /*  Event Handlers */
 
-function handleProfileEditSubmit(data) {
+function avatarSubmit(link) {
+  avatarProfilePopup.modalLoading(true);
+  api
+    .updateAvatar(link)
+    .then((res) => {
+      console.log(res);
+      userInfo.updateProfileImage(res.avatar);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      avatarProfilePopup.modalLoading(false);
+    });
+  avatarProfilePopup.close();
+  avatarFormElement.reset();
+}
```

Comment:
These 2 lines should be in `then`

* You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well


You need to **reset** the form only   in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.

**Please, correct it everywhere**